### PR TITLE
`BlockchainBuilder` abstraction to simplify instantiating chains

### PR DESCRIPTION
**Pull request by neysofu**
NEAR, Cosmos and Arweave all share the same `Chain::new` type signature. By putting that method into a trait, we can reduce code duplication in the `_networks_as_chains` family of functions in `main.rs`.

### CODE REVIEW

**Review by mangas — APPROVED** [review]
Cool, I'm going to try and do something similar to block ingestors and block streams since all they need now is ChainClient :)

**Comment by neysofu** [comment]
@mangas I went a step further after your review; I completely removed `Chain::new` and rather instantiate directly in `Chain::build`. WDYT? See https://github.com/graphprotocol/graph-node/pull/4408/commits/8d29da588225419af549d0131c5a5ee075456cad.

File: chain/arweave/src/chain.rs [hunk 1]
```diff
@@ -44,20 +44,20 @@ impl std::fmt::Debug for Chain {
     }
 }
 
-impl Chain {
-    pub fn new(
+impl BuildableBlockchain for Chain {
+    fn build(
         logger_factory: LoggerFactory,
-        name: String,
+        chain_id: String,
```
Comment (mangas) [inline chain/arweave/src/chain.rs:50]: This is the network_name, there is also a chain_id whihc is something internal to each chain, I think we should keep the previous naming

File: chain/arweave/src/chain.rs [hunk 2]
```diff
@@ -44,20 +44,20 @@ impl std::fmt::Debug for Chain {
...
         chain_store: Arc<dyn ChainStore>,
-        firehose_endpoints: FirehoseEndpoints,
-        metrics_registry: Arc<dyn MetricsRegistry>,
+        fh_endpoints: FirehoseEndpoints,
```
Comment (mangas) [inline chain/arweave/src/chain.rs:52]: please leave this one as was, I have a PR that will replace FIrehoseEndpoints with ChainClient in most places

File: graph/src/blockchain/mod.rs [hunk 3]
```diff
@@ -135,6 +139,19 @@ impl ChainStoreBlock {
 //     Rpc(C::Client),
 // }
 
+/// A chain-agnostic interface for instantiating chains of a specific kind. Not
+/// all chains may support this interface, but that's okay, as some may require
+/// special logic.
+pub trait BuildableBlockchain: Blockchain {
```
Comment (mangas) [inline graph/src/blockchain/mod.rs:145]: This comment is totally subjective so I will leave it just for your consideration.

I think I would rather have a the Blockchain as a type parameter. 

Also have a suggestion for the name 

```
pub trait BlockchainBuilder<C: Blockchain>
```

This means we could have multiple implementations of Builders if needed or a single one that's generic over Blockchain as opposed to having only 1 per Blockchain implementation. 

Other than that I don't have a huge preference and may also be missing something here.

**Review by mangas — COMMENTED** [review]
Left a few more comments but in general LGTM

**Comment by neysofu** [comment]
I like the `BlockchainBuilder` pattern much better, thank you for the suggestion!

**Pull request merged**